apps/hash-graph/lib/graph/src/shared/identifier/time/temporal_axes.rs

```diff
@@ -0,0 +1,341 @@
+use serde::{Deserialize, Serialize};
+use utoipa::{openapi, ToSchema};
+
+use crate::identifier::time::{
+    axis::{PinnedAxis, TemporalTagged},
+    bound::TemporalBound,
+    DecisionTime, LimitedTemporalBound, LimitedTemporalInterval, TemporalInterval, TimeAxis,
+    Timestamp, TransactionTime, UnresolvedTemporalInterval, VariableAxis, VersionInterval,
+};
+
+#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct UnresolvedPinnedTemporalAxis<A> {
```

Comment:
Do we want to add docs for all of these? It shouldn't be hard to adapt from the BP implementation's docs

Reply:
I copied over and adjusted the docs from BP in [Add documentation from BP specs](https://github.com/hashintel/hash/pull/1854/commits/3dfd2b01a8132bc8009de26a2809181382af34d2)